protocols/tcp/iscsi.go

```diff
@@ -0,0 +1,144 @@
+package tcp
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"fmt"
+	"log/slog"
+	"net"
+
+	"github.com/mushorg/glutton/connection"
+	"github.com/mushorg/glutton/producer"
+	"github.com/mushorg/glutton/protocols/interfaces"
+)
+
+type iscsiMsg struct {
+	Opcode  uint8
+	Flags   uint8
+	TaskTag uint32
+	Data    uint32
+	CID     uint32
+	LUN     uint64
+}
+
+type iscsiRes struct {
+	Opcode  uint8
+	Flags   uint8
+	TaskTag uint32
+	Data    uint32
+	CID     uint32
+	LUN     uint64
+	Status  uint8
+}
+
+type iscsiRequest struct {
+	Msg iscsiMsg
+	Res iscsiRes
+}
+
+type iscsiServer struct {
+	events []iscsiRequest
```

Comment:
[This](https://github.com/mushorg/glutton/blob/52e236f8e1fcc1109b126cfca921eb24379c7d1f/protocols/tcp/bittorrent.go#L24) is how the event should look like